### PR TITLE
Support multiple dist configurations and ESM

### PR DESCRIPTION
**Pull request by ardatan**

### CODE REVIEW

File: src/commands/build.ts [hunk 1]
```diff
@@ -361,18 +390,18 @@ export function validatePackageJson(pkg: any) {
...
-  expect("typescript.definition", `${distDir}/index.d.ts`);
+  expect("main", `${DIST_DIR}/index.cjs.js`);
+  expect("module", `${DIST_DIR}/index.esm.js`);
+  expect("typings", `${DIST_DIR}/index.d.ts`);
+  expect("typescript.definition", `${DIST_DIR}/index.d.ts`);
 
   if (pkg.exports) {
     expect("exports.require", pkg.main);
-    expect("exports.default", `${distDir}/index.mjs`);
+    expect("exports.default", `${DIST_DIR}/index.mjs`);
```
Comment (n1ru4l) [inline src/commands/build.ts:400]: I noticed that VSCode and Node12 started complaining about "exports.default" and "exports.require" in the package.json of the output package. It must be prefixed with `./`. However, this started to appear on my WIndows machine, have not checked on osx yet. After I prefixed it, bob started complaining here. 

```
Error: @n1ru4l/graphql-live-query: "exports.require" equals "./dist/index.cjs.js", should be "dist/index.cjs.js"
    at expect (/home/runner/work/graphql-live-queries/graphql-live-queries/node_modules/bob-the-bundler/dist/commands/build.js:265:19)
    at validatePackageJson (/home/runner/work/graphql-live-queries/graphql-live-queries/node_modules/bob-the-bundler/dist/commands/build.js:273:9)
    at build (/home/runner/work/graphql-live-queries/graphql-live-queries/node_modules/bob-the-bundler/dist/commands/build.js:117:5)
error Command failed with exit code 1.
```

So I commented this assertion out... It worked afterward. The examples here are also all prefixed with `./`

Comment (ardatan) [reply src/commands/build.ts]: Fixed!

File: src/config.ts [hunk 2]
```diff
@@ -19,6 +19,10 @@ export interface BobConfig {
   commands?: {
     [cmdName: string]: Command;
   };
+  dists?: {
+    distDir: string;
+    distPath?: string;
+  }[]
```
Comment (kamilkisiela) [inline src/config.ts:25]: Do we need this or could we maybe have `cjs`, `esm`, `nodeEsm` flags? @ardatan

Comment (kamilkisiela) [reply src/config.ts:25]: what's the story behind `dists`?

Comment (ardatan) [reply src/config.ts:25]: Having es5 builds for graphql-tools
https://github.com/ardatan/graphql-tools/blob/master/bob.config.js#L9
We have needed that before in MSTeams

Comment (ardatan) [reply src/config.ts:25]: Yes it is actually. https://github.com/ardatan/graphql-tools/blob/master/package.json#L10
Here we run tsc twice in parallel, one is for modern es and the other one is for es5.

Comment (ardatan) [reply src/config.ts:25]: I am not sure this old solution is best but it worked for MSTeams :D It was a long time ago, we can change it for sure. But that was the reason behind having this `dists` thing.

Comment (kamilkisiela) [reply src/config.ts:25]: @kamilkisiela @kamilkisiela (for me) add nodeEsm flag to the PR and merge it

Comment (kamilkisiela) [reply src/config.ts:25]: (also for me) update validation rules in bob validate to make sure it’s in sync

File: src/commands/validate.ts [hunk 3]
```diff
@@ -163,14 +163,14 @@ export async function validatePackage({
...
-    `${name}: 'main' should equal 'dist/index.cjs.js'`,
+    "dist/index.js",
+    `${name}: 'main' should equal 'dist/index.js'`,
     onError
   );
   shouldEqual(
     pkg.module,
-    "dist/index.esm.js",
-    `${name}: 'module' should equal 'dist/index.esm.js'`,
+    "dist/index.msj",
```
Comment (ardatan) [inline src/commands/validate.ts:172]: I think this is typo :D

Comment (kamilkisiela) [reply src/commands/validate.ts:172]: I think so too! :D

**Pull request merged**